access-protected-api/README.md

```diff
@@ -0,0 +1,162 @@
+| **Client secret**              | _**Value** of the client secret (not its ID)_                        | :warning: Record this value immediately! <br/> It's shown only _once_ (when you create it). |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+Open the [_application.yml_](src/main/resources/application.yml) file and modify the four Azure Active Directory configuration properties using the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-expose-web-apis).
+
+```yaml
+# 'Tenant ID' of your Azure AD instance - this value is a GUID
+tenant-id: value-here
+
+# 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+client-id: value-here
+
+# Client secret 'Value' (not its ID) from 'Client secrets' in app registration in Azure portal
+client-secret: value-here
+
+# 'Object ID' of app registration in Azure portal - this value is a GUID
+client-object-id: value-here
+```
+
+### 3. Install package(s) and compile application
+
+To install Spring Boot and MSAL libraries:
+
+```bash
+mvn install
+```
+
+## Run the application
+
+```bash
+mvn spring-boot:run
+```
+
+## Access the API
+
+Using Postman, curl, or a similar application, issue an HTTP GET request to *http://localhost:8080/api/application*.
+
+For example, if you use curl and everything worked, the sample you should receive a response from the API similar to this:
+
+```console
+$ curl http://localhost:8080/api/application
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#applications/$entity",
+  "id": "04487cab-91cf-46d8-b4ed-1a428388bdd2",
+  "appId": "de5dcdc6-8dfe-4048-aff6-c615dea4ee3d",
+  "displayName": "java-api",
+  "description": null,
+  "groupMembershipClaims": null,
+  "identifierUris": ["api://de5dcdc6-8dfe-4048-aff6-c615dea4ee3d"],
+  "isDeviceOnlyAuthSupported": null,
+  "isFallbackPublicClient": null,
+  "notes": null,
+  "publisherDomain": "contoso.onmicrosoft.com",
+  "signInAudience": "AzureADMyOrg",
+  "tags": [],
+  "tokenEncryptionKeyId": null,
+  "defaultRedirectUri": null,
+  "certification": null,
+  "optionalClaims": null,
+  …
+}
+```
+
+## About the code
+
+This Java API uses the Spring Boot web framework. The app has a single, anonymous route. When that route is called, the application will make a Microsoft Graph request to get information about its own app registration. This is being done via the client credentials OAuth flow, meaning the application is using its own App registration as the credentials to generate the access token for Microsoft Graph. The results of the Microsoft Graph call are then returned as the results of the API call.

```

Comment:
```suggestion
This Java web API uses the Spring Boot web framework and has a single route that supports anonymous access. When its anonymous route is called, the API requests its own application object from the Microsoft Graph API.

The web API uses MSAL for Java to get an access token for Microsoft Graph by using the OAuth 2.0 client credentials flow. That is, this web API (a confidential client) uses the credentials in its app registration (and thus its own identity) when requesting the access token for Microsoft Graph. If its call to the Microsoft Graph API was successful, the Java web API then returns the results to its anonymous caller.
```

I'm not sure if this is any clearer but felt like noodling on it. It's a Friday.

Reply:
I dig it.